### PR TITLE
Remove redundant DestinationRule

### PR DESCRIPTION
**Pull request by Tomasz-Smelcerz-SAP**
**Description**

As https://github.com/kyma-project/kyma/pull/11600 is apparently not working, why not just remove the apparently redundant DestinationRule?

Details:
a DestinationRule for K8s API server was embedded in Istio sources up to 1.4.x., and our rule reflects the built-in one.
In 1.5.x it's installation in Istio was conditional, it wasn't installed if "auto-mtls" is used - which was "on" by default.
In later Istio versions there's no such DestinationRule at all.
I think it's a legacy object that can be safely deleted.

Changes proposed in this pull request:

- remove DestinationRule for K8s API server from Kyma sources
- remove questionable statement from the documentation

**Related issue(s)**
#11478

### CODE REVIEW

File: docs/service-mesh/03-01-istio-setup-in-kyma.md [hunk 1]
```diff
@@ -33,7 +33,7 @@ These configuration changes are applied to customize Istio for use with Kyma:
 - [Mutual TLS (mTLS)](https://istio.io/docs/concepts/security/#mutual-tls-authentication) is enabled cluster-wide in a STRICT mode.
 - Global tracing is set to use the Zipkin installation provided by Kyma.
 - Ingress Gateway is expanded to handle ports `80` and `443` for local Kyma deployments.
-- DestinationRules are created by default, which disables mTLS for the `kubernetes.default.svc.cluster.local` service. In local (Minikube) installation mTLS is also disabled for
+- In local (Minikube) installation there is a DestinationRule that disables mTLS for
```
Comment (mmitoraj) [inline docs/service-mesh/03-01-istio-setup-in-kyma.md:36]:
```suggestion
- In local (Minikube) installation there is a DestinationRule that disables mTLS for the `istio-ingressgateway.istio-system.svc.cluster.local` service.
```

File: docs/service-mesh/03-01-istio-setup-in-kyma.md [hunk 2]
```diff
@@ -33,7 +33,7 @@ These configuration changes are applied to customize Istio for use with Kyma:
...
 `istio-ingressgateway.istio-system.svc.cluster.local` service.
```
Comment (mmitoraj) [inline docs/service-mesh/03-01-istio-setup-in-kyma.md:37]: We can move that piece to the line above as it's one sentence. See the above comment.

**Review by mmitoraj — CHANGES_REQUESTED** [review]
I added two small suggestons. Thanks!

**Review by cnvergence — DISMISSED** [review]

**Review by mmitoraj — APPROVED** [review]

**Comment by cnvergence** [comment]
LGTM, this destination rule wasn't doing the thing described in the comments

**Review by cnvergence — APPROVED** [review]

**Review by mjakobczyk — APPROVED** [review]

**Pull request merged**